tests/profiler/test_profiler.py

```diff
@@ -343,16 +348,16 @@ def test_pytorch_profiler_trainer_fit(fast_dev_run, boring_model_cls, tmpdir):
     trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, fast_dev_run=fast_dev_run, profiler=pytorch_profiler)
     trainer.fit(model)
 
-    assert sum(e.name == "validation_step" for e in pytorch_profiler.function_events)
+    assert sum(
+        e.name == "TrainingTypePlugin.SingleDevicePlugin.validation_step" for e in pytorch_profiler.function_events
```

Comment:
what is the motivation behind this pattern of including the base class? Wouldn't it have to be separated by a different symbol?

Reply:
I also find it confusing as it's not something that exists. I suggest removing the base class

Reply:
see @ananthsub's comments above

Reply:
@awaelchli @carmocca this would be useful for automatically categorizing profiler data, especially for determining what came from the module vs callback vs strategy

Reply:
I understand that, but the current structure makes it sound like `TrainingTypePlugin.SingleDevicePlugin.validation_step` is an attribute that exists but that's correct in Python.

I see two options:
- Drop the base class
- Remove the ambiguity with different notation, for example `[TrainingTypePlugin] SingleDevicePlugin.validation_step`

Reply:
notation with brackets sounds good to me

Reply:
Sg, will update!